### PR TITLE
Zookeeper

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: container/karaf/pom.xml [hunk 1]
```diff
@@ -35,9 +35,10 @@
     <packaging>pom</packaging>
 
     <modules>
-        <module>distro</module>
         <module>managed</module>
+        <module>features</module>
         <module>patch</module>
+        <module>distro</module>
```
Comment (tdiesler) [inline container/karaf/pom.xml]: module ordering should be defined by module dependencies - here we want to have alphabetical ordering

File: container/wildfly/patch/pom.xml [hunk 2]
```diff
@@ -119,7 +172,7 @@
                             <ignoreDirFormatExtensions>true</ignoreDirFormatExtensions>
                             <appendAssemblyId>false</appendAssemblyId>
                             <descriptors>
-                                <descriptor>scripts/assembly-deploy-artifacts.xml</descriptor>
+                                <descriptor>src/main/scripts/assembly-deploy-artifacts.xml</descriptor>
```
Comment (tdiesler) [inline container/wildfly/patch/pom.xml:175]: has the script location consistently been changed in all modules?

Comment (iocanel) [inline container/wildfly/patch/pom.xml:175]: No, only in that module.

File: fabric/core/src/main/java/io/fabric8/core/internal/curator/Config.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ * See the License for the specific language governing permissions and
+ */
+package io.fabric8.core.internal.curator;
+
+import static io.fabric8.core.internal.curator.Constants.DEFAULT_CONNECTION_TIMEOUT_MS;
+import static io.fabric8.core.internal.curator.Constants.DEFAULT_RETRY_INTERVAL;
+import static io.fabric8.core.internal.curator.Constants.DEFAULT_SESSION_TIMEOUT_MS;
+import static io.fabric8.core.internal.curator.Constants.MAX_RETRIES_LIMIT;
+
+public class Config {
```
Comment (tdiesler) [inline fabric/core/src/main/java/io/fabric8/core/internal/curator/Config.java]: does this really need to be public?
same for comment for all public methods

File: container/karaf/pom.xml (hunk 1 above)
Comment (iocanel) [inline container/karaf/pom.xml]: Ok, I'll have the distro depend on the features then.

File: fabric/core/src/main/java/io/fabric8/core/internal/curator/Config.java (hunk 3 above)
Comment (iocanel) [inline fabric/core/src/main/java/io/fabric8/core/internal/curator/Config.java]: No.

File: fabric/core/src/main/java/io/fabric8/core/internal/curator/CuratorFrameworkLocator.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import io.fabric8.spi.scr.ValidatingReference;
+import org.apache.curator.framework.CuratorFramework;
+
+/**
+ * Locate the {@link org.apache.curator.framework.CuratorFramework}
+ *
+ * @since 10-Dec-2013
+ */
+public final class CuratorFrameworkLocator {
```
Comment (tdiesler) [inline fabric/core/src/main/java/io/fabric8/core/internal/curator/CuratorFrameworkLocator.java:25]: why would this be needed? All client API goes through services, which should have dependencies

File: fabric/core/src/main/java/io/fabric8/core/internal/utils/Base64Encoder.java [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+package io.fabric8.core.internal.utils;
+
+import java.nio.charset.Charset;
+
+/**
+ * Bease64 encoding utilities.
+ *
+ * @since 0.5.0, January 16, 2008
+ */
+public class Base64Encoder {
```
Comment (tdiesler) [inline fabric/core/src/main/java/io/fabric8/core/internal/utils/Base64Encoder.java]: already exists in gravia

File: fabric/core/src/main/java/io/fabric8/core/internal/utils/Closeables.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+
+package io.fabric8.core.internal.utils;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+public final class Closeables {
```
Comment (tdiesler) [inline fabric/core/src/main/java/io/fabric8/core/internal/utils/Closeables.java]: already exists in gravia IOUtils.safeClose(...)

File: itests/smoke/embedded/src/main/java/io/fabric8/test/embedded/support/EmbeddedTestSupport.java [hunk 7]
```diff
@@ -36,7 +36,9 @@
  */
 public abstract class EmbeddedTestSupport {
 
-    private static String[] moduleNames = new String[] { "gravia-provision", "gravia-resolver", "gravia-repository", "fabric8-api", "fabric8-spi", "fabric8-core" };
+    private static String[] moduleNames = new String[] { "gravia-provision", "gravia-resolver", "gravia-repository",
+            "fabric8-api", "fabric8-spi",
+            "zookeeper", "guava", "curator-client", "curator-framework", "fabric8-core" };
```
Comment (tdiesler) [inline itests/smoke/embedded/src/main/java/io/fabric8/test/embedded/support/EmbeddedTestSupport.java]: why is guava in this list?

File: pom.xml [hunk 8]
```diff
@@ -61,6 +61,8 @@
         <version.apache.felix.scr.generator>1.8.0</version.apache.felix.scr.generator>
         <version.apache.karaf>2.3.3</version.apache.karaf>
         <version.apache.tomcat>7.0.42</version.apache.tomcat>
+        <version.curator>2.4.2</version.curator>
```
Comment (tdiesler) [inline pom.xml]: version.apache.curator

File: pom.xml [hunk 9]
```diff
@@ -74,6 +76,7 @@
         <version.mockito>1.8.4</version.mockito>
         <version.osgi>5.0.0</version.osgi>
         <version.slf4j>1.6.6</version.slf4j>
+        <version.zookeeper>3.4.6</version.zookeeper>
```
Comment (tdiesler) [inline pom.xml]: version.apache.zookeeper

File: fabric/core/src/main/java/io/fabric8/core/internal/curator/CuratorFrameworkLocator.java (hunk 4 above)
Comment (iocanel) [inline fabric/core/src/main/java/io/fabric8/core/internal/curator/CuratorFrameworkLocator.java:25]: For use in short-lived objects with lifecycle not managed by us (e.g. a LoginModule).

File: itests/smoke/embedded/src/main/java/io/fabric8/test/embedded/support/EmbeddedTestSupport.java (hunk 7 above)
Comment (iocanel) [inline itests/smoke/embedded/src/main/java/io/fabric8/test/embedded/support/EmbeddedTestSupport.java]: Guava is a dependency of Curator. I think that in the latest version has been removed, so we may be able to completely wipe guava.

File: pom.xml (hunk 8 above)
Comment (iocanel) [inline pom.xml]: apache is redudant.

**Pull request merged**